pyEpiabm/pyEpiabm/sweep/demographics_sweep.py

```diff
@@ -0,0 +1,99 @@
+#
+# Sweep to record demographic information of the population in a .csv file
+#
+
+import os
+import typing
+
+from pyEpiabm.output import _CsvDictWriter
+
+from .abstract_sweep import AbstractSweep
+
+
+class DemographicsSweep(AbstractSweep):
+    """Class to sweep through the population at the beginning of the simulation
+    and record their demographic information
+
+    """
+
+    def __init__(self, file_params: typing.Dict):
+        """Initiate file parameters referring to the file location and
+        which columns are to be included
+
+        file_params Contains:
+            * `output_dir`: String for the location of the output file, \
+               as a relative path
+            * `spatial_output`: Boolean to determine whether a spatial output \
+               should be used
+            * `age_output`: Boolean to determine whether the output will \
+                have ages
```

Comment:
Somewhere you should check that `use_ages` in the params is actually true if this is true, and log an error if not. Not much point in an age stratified output if the simulation isn't age stratified! Alternatively, given the minimal extra cost, I would happy with you adding age data by default if use_ages is true.

Reply:
Ah yes, this makes sense, I will add checks for use_ages